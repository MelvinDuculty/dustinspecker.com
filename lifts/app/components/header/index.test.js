import Header from '.'
import {MemoryRouter} from 'react-router'
import React from 'react'
import renderer from 'react-test-renderer'

test('Header displays "Lifts" when no lift selected', () => {
  const props = {
    lifts: []
  }

  const component = renderer.create(
    <MemoryRouter>
      <Header {...props} />
    </MemoryRouter>
  )

  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

test('Header displays "Lifts | Name" when lift selected', () => {
  const props = {
    lifts: [
      {
        id: '2',
        name: 'first lift'
      },
      {
        id: '4',
        name: 'second lift'
      },
      {
        id: '5',
        name: 'third lift'
      }
    ]
  }

  const memoryRouterProps = {
    initialEntries: ['/4']
  }

  const component = renderer.create(
    <MemoryRouter {...memoryRouterProps} >
      <Header {...props} />
    </MemoryRouter>
  )

  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
