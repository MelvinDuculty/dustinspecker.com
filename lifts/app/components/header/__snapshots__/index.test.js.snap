// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header displays "Lifts | Name" when lift selected 1`] = `
<header
  className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-8"
>
  <div
    className="MuiToolbar-root-37 MuiToolbar-gutters-38 Header-toolbar-1"
  >
    <h2
      className="MuiTypography-root-39 MuiTypography-title-45 MuiTypography-colorInherit-58"
    >
      Lifts | Second Lift
    </h2>
    <a
      className="MuiButtonBase-root-78 MuiButton-root-63 MuiButton-colorInherit-67"
      href="#/"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-64"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-81"
        >
          home
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-87"
      />
    </a>
  </div>
</header>
`;

exports[`Header displays "Lifts" when no lift selected 1`] = `
<header
  className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-8"
>
  <div
    className="MuiToolbar-root-37 MuiToolbar-gutters-38 Header-toolbar-1"
  >
    <h2
      className="MuiTypography-root-39 MuiTypography-title-45 MuiTypography-colorInherit-58"
    >
      Lifts
    </h2>
    <a
      className="MuiButtonBase-root-78 MuiButton-root-63 MuiButton-colorInherit-67"
      href="#/"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-64"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-81"
        >
          home
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-87"
      />
    </a>
  </div>
</header>
`;
